browser_history/generic.py

```diff
@@ -562,25 +562,32 @@ def bookmarks_parser(self, bookmark_path):
         :rtype: list(tuple(:py:class:`datetime.datetime`, str, str, str))
         """
 
-        def _deeper(array, folder, bookmarks_list):
-            for node in array:
-                if node["type"] == "url":
-                    d_t = datetime.datetime(1601, 1, 1) + datetime.timedelta(
-                        microseconds=int(node["date_added"])
-                    )
-                    bookmarks_list.append(
-                        (
-                            d_t.replace(microsecond=0, tzinfo=self._local_tz),
-                            node["url"],
-                            node["name"],
-                            folder,
-                        )
-                    )
+        def _deeper(bookmarks_json, folder, bookmarks_list):
+            for node in bookmarks_json:
+                if node == "children":
+                    for child in bookmarks_json[node]:
+                        if child["type"] == "url":
+                            d_t = datetime.datetime(1601, 1, 1) + datetime.timedelta(
+                                microseconds=int(child["date_added"])
+                            )
+                            bookmarks_list.append(
+                                (
+                                    d_t.replace(microsecond=0, tzinfo=self._local_tz),
+                                    child["url"],
+                                    child["name"],
+                                    folder,
+                                )
+                            )
+                        else:
+                            bookmarks_list = _deeper(
+                                child,
+                                folder + os.sep + child["name"],
+                                bookmarks_list,
+                            )
```

Comment:
This should be done when `type` is `folder`, right? Maybe you could use a `elif` instead of `else`?

Reply:
Yea sure
Your concern here would be understanding of the code right?
or is there any other reason?

Reply:
My concern is that if the format changes and there's another `type` besides `folder`, it might error out

Reply:
Right!,